### DISCUSSION TITLE
How to add 1-5px border to macos

### DISCUSSION THREAD

**Discussion by piotr-sikora-v** [kovidgoyal/kitty#7696] [Q&A] [answered]
I have config:

```
hide_window_decorations yes
window_border_width 5px
draw_minimal_borders no
kitty_mod ctrl+shift
clear_all_shortcuts no
#map cmd+enter new_os_window
map cmd+enter launch /Applications/kitty.app/Contents/MacOS/kitty -1 
macos_quit_when_last_window_closed no
copy_on_select yes
shell_integration disabled
```

but `window_border_width` not working :/

![Screenshot 2024-08-04 at 08 27 58](https://github.com/user-attachments/assets/c1560bf2-0a5f-4e33-b5be-ca360dabfc66)

**Comment by kovidgoyal** [comment, marked as answer]
Works fine for me with

kitty --config=NONE -o window_border_width=20px -o draw_minimal_borders=no -o enabled_layouts=tall

**Reply by piotr-sikora-v** [reply]
Thansk... problem was with hide_window_decorations.
But now I try to remove rounded corners and when I set hide_window_decorations to `titlebar-and-corners` then i'am backign to same thing as on screenshot :/

```
kitty --config=none -o window_border_width=10px -o draw_minimal_borders=no -o hide_window_decorations=titlebar-and-corners
```

**Comment by kovidgoyal** [comment]
hide_window_decorations is about controlling the appearance of the OS
Window. window_border is about controlling the border around kitty
windows, of which there can be multiple per OS Window. window borders
around kitty windows are only drawn when there is more than one in a
given OS Window.

**Reply by piotr-sikora-v** [reply]
some option to draw tiny borders on macos without rounded corners and without title?

**Comment by kovidgoyal** [comment]
On Sun, Aug 04, 2024 at 11:34:10PM -0700, Piotr Sikora wrote:
> some option to draw tiny borders on macos without rounded corners and without title?

OS window borders are drawn by the OS not kitty, there is no way to
control them beyond turning parts of them on and off.
